wsclient/src/main/java/org/hpccsystems/ws/client/BaseHPCCWsClient.java

```diff
@@ -88,6 +102,32 @@ public boolean isTargetHPCCContainerized() throws Exception
         return targetsContainerizedHPCC;
     }
 
+    @WithSpan
+    private boolean getTargetHPCCAuthenticates(Connection conn) throws Exception
+    {
+        if (wsconn == null)
+            throw new Exception("Cannot get target HPCC authentication mode, client connection has not been initialized.");
+
+        String response = wsconn.sendGetRequest("esp/getauthtype");//throws
+
+        if (response == null || response.isEmpty())
+            throw new Exception("Cannot get target HPCC authentication mode, received empty " + wsconn.getBaseUrl() + " esp/getauthtype response");
+
+        /*
+         * Known auth types:
+         * static const char* const AUTH_TYPE_NONE = "None";
+         * static const char* const AUTH_TYPE_USERNAMEONLY = "UserNameOnly";
+         * static const char* const AUTH_TYPE_PERREQUESTONLY = "PerRequestOnly";
+         * static const char* const AUTH_TYPE_PERSESSIONONLY = "PerSessionOnly";
+         * static const char* const AUTH_TYPE_MIXED = "Mixed";
+         * Declared in esp/platform/espcontext.hpp#L62
+         */
+        if (response.indexOf("None") > 0) //<GetAuthTypeResponse><AuthType>None|Mixed</AuthType></GetAuthTypeResponse>
```

Comment:
This seems unlikely to change, but should we be doing case insensitive comparison here?

Reply:
yeah I considered it, but confirmed the only possible casing.